client/src/components/Atoms/StatusTag.tsx

```diff
@@ -1,9 +1,36 @@
+import React from 'react'
 import styled from 'styled-components'
-import { Tag } from '@blueprintjs/core'
+import { Tag, ProgressBar, ITagProps } from '@blueprintjs/core'
 
 const StatusTag = styled(Tag).attrs({ minimal: true })`
   text-transform: uppercase;
   font-weight: 500;
 `
 
+const StatusTagWithProgressBar = styled(StatusTag)`
+  position: relative;
+  border-bottom-left-radius: 0;
+  border-bottom-right-radius: 0;
+
+  .bp3-progress-bar {
+    position: absolute;
+    bottom: -2px;
+    left: 0;
+    border-radius: 0;
+    height: 2px;
+  }
+  .bp3-progress-meter {
+    border-radius: 0;
+  }
+`
+
+export const StatusTagWithProgress: React.FC<{
+  progress: number
+} & Omit<ITagProps, 'minimal'>> = ({ progress, children, ...props }) => (
+  <StatusTagWithProgressBar {...props}>
+    {children}
+    <ProgressBar stripes={false} value={progress} intent={props.intent} />
+  </StatusTagWithProgressBar>
+)
```

Comment:
I imagine in most cases where we'll use a `StatusTagWithProgress`, we'll swap it out with a plain `StatusTag` once the relevant work is done and progress has hit 100%. Rather than exporting two components that consumers have to toggle, would it make sense to export a single `StatusTag` component with a `progress: number | undefined` prop that renders no progress bar when `progress` is `undefined`?

Reply:
I briefly explored that. I can't remember exactly why I abandoned it and decided to stick with separate components. I'll give it another quick look.

Reply:
Ok it was a bit hairy but I pulled it off: 8e215f59